src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -211,4 +230,135 @@
         r.assertBuildStatusSuccess(r.waitForCompletion(b4));
     }
 
+    @Issue("JENKINS-34005")
+    @Test public void triggersProperty() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        // Verify the base case behavior.
+        p.setDefinition(new CpsFlowDefinition("echo 'foo'"));
+
+        assertTrue(p.getTriggers().isEmpty());
+
+        r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+        // Make sure the triggers are still empty.
+        assertTrue(p.getTriggers().isEmpty());
+
+        // Now add a trigger.
+        p.setDefinition(new CpsFlowDefinition(
+                (HAVE_SYMBOL ?
+                        "properties([pipelineTriggers([\n"
+                                + "  cron('@daily'), [$class: 'MockTrigger']])])\n" :
+                        "properties([pipelineTriggers(\n"
+                                + "  triggers: [[$class: 'TimerTrigger', spec: '@daily'],\n"
+                                + "    [$class: 'MockTrigger']])])\n"
+                ) + "echo 'foo'"));
+
+        WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+        // Verify that we're seeing warnings due to running 'properties' in a non-multibranch job.
+        r.assertLogContains("WARNING: The 'properties' step will remove all 'JobProperty's currently configured in this job, either from the UI or from an earlier 'properties' step.", b);
```

Comment:
:ant: move this into `Messages.properties` and reuse the method call in the test, rather than copying & pasting a long and detailed message which we might well want to tweak.